internal/e2e/main_test.go

```diff
@@ -174,32 +168,34 @@ func Setup(t testing.TB) TestData {
 		err = nil
 	}
 	// cannot delete policies until all identities are detached
-	var pes madmin.PolicyEntitiesResult
-	if le {
-		pes, err = ma.GetLDAPPolicyEntities(ctx, madmin.PolicyEntitiesQuery{Policy: []string{policy.Spec.Name}})
-	} else {
-		pes, err = ma.GetPolicyEntities(ctx, madmin.PolicyEntitiesQuery{Policy: []string{policy.Spec.Name}})
+	lpes, err := ma.GetLDAPPolicyEntities(ctx, madmin.PolicyEntitiesQuery{Policy: []string{policy.Spec.Name}})
```

Comment:
apparently, policy attachments will always prevent policy deletion - even if an identity provider is set that prevents the policy attachments from being enforced.  so, if you have builtin identity attachments, but have an ldap identity provider enabled, these attachments will still prevent deletion.

this section makes policy cleanup more robust between e2e tests.